File: java/src/test/java/mx/iteso/adapter/GoogleDocAdapterTest.java
package mx.iteso.adapter;

import mx.iteso.adapter.googledoc.GoogleDoc;
import mx.iteso.adapter.googledoc.GoogleDocAdapter;
import mx.iteso.adapter.word.MSLicense;
import org.junit.Before;
import org.junit.Test;

import static org.junit.Assert.assertEquals;
import static org.junit.Assert.assertTrue;

public class GoogleDocAdapterTest {
    private GoogleDocAdapter gAdapter;
    @Before
    public void init() {
        gAdapter = new GoogleDocAdapter(new GoogleDoc());
    }
    @Test
    public void backgroundTest() {
        Image gAdapterBackground = gAdapter.getBackground();
        assertEquals(400.0, gAdapterBackground.getWidth(), 0.0);
        assertEquals(600.0, gAdapterBackground.getHeight(), 0.0);
    }
    @Test
    public void licenseTest() {
        MSLicense gAdapterLicense = gAdapter.getLicense();
        assertEquals(2.0, gAdapterLicense.getLicenseVersion(), 0.0);
        assertTrue(gAdapter.restrictEditIfLicenseIsInvalid());
    }
    @Test
    public void formatTest() {
        Format gAdapterFomart = gAdapter.getFormat();
        assertEquals("TimesNewRoman", gAdapterFomart.getFont().getFontName());
        assertEquals(25, gAdapterFomart.getFont().getFontSize());

    }
}

File: python/src/unittest/python/GoogleDocAdapterTest.py
import unittest
from src.main.python.Documents import GoogleDoc
from src.main.python.Adapters import GoogleDocAdapter


class GoogleDocAdapterTest(unittest.TestCase):

    def setUp(self):
        self.word_doc_adapter = GoogleDocAdapter(GoogleDoc())

    def test_format(self):
        w_format = self.word_doc_adapter.get_format()
        self.assertEqual("TimesNewRoman", w_format.get_font().get_font_name())
        self.assertEqual(25, w_format.get_font().get_font_size())

    def test_background(self):
        background = self.word_doc_adapter.get_background()
        self.assertEqual(400.0, background.get_width(), 0.0)
        self.assertEqual(600.0, background.get_height(), 0.0)

    def test_license(self):
        license = self.word_doc_adapter.get_license()
        self.assertEqual(2.0, license.get_license_version())


if __name__ == '__main__':
    unittest.main()
